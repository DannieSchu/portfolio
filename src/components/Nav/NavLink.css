.NavLink {
  cursor: pointer;
  height: 4rem;

  img {
    width: 3.5rem;
  }

  img.solid, p {
    display: none;
  }

  &:hover {
    img {
      display: none;
    }

    p {
      display: block;
      height: 100%;
    }
  }
}

.active {
  img.solid {
    display: block;
  }

  img.outline, p {
    display: none;
  }

  &:hover {
    img.solid {
      display: none;
    }

    p {
      display: block;
      height: 100%;
    }
  }
}
