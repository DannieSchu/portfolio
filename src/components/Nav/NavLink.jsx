import React from 'react';
import { Link } from 'react-scroll';
import PropTypes from 'prop-types';
import styles from './NavLink.css';
import typeStyle from '../../styles/type.css';

const NavLink = ({ to, text, icon, activeIcon }) => (
  <Link
    to={to}
    spy={true}
    smooth={true}
    duration={800}
    activeClass={styles.active}
    className={styles.NavLink}>
    <img src={icon} alt={to} className={styles.outline} />
    <img src={activeIcon} alt={`${to} active`} className={styles.solid} />
    <p className={typeStyle.allCaps}>{text}</p>
  </Link>
);

NavLink.propTypes = {
  to: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  activeIcon: PropTypes.string.isRequired,
};

export default NavLink;
