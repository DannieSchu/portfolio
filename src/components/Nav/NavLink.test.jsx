import React from 'react';
import { shallow } from 'enzyme';
import NavLink from './NavLink';

describe('NavLink component', () => {
  it('renders NavLink', () => {
    const wrapper = shallow(
      <NavLink to="Home" icon="home.png" activeIcon="home-active.png" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
