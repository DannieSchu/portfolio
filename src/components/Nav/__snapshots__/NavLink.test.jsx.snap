// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink component renders NavLink 1`] = `
<Link
  activeClass="active"
  className="NavLink"
  duration={800}
  offset={0}
  smooth={true}
  spy={true}
  to="Home"
>
  <img
    alt="Home"
    className="outline"
    src="home.png"
  />
  <img
    alt="Home active"
    className="solid"
    src="home-active.png"
  />
  <p
    className="allCaps"
  />
</Link>
`;
